### PR TITLE
review

### PR DESCRIPTION
**Pull request by vvo**

### CODE REVIEW

File: scripts/docs/metalsmith/content/concepts.md [hunk 1]
```diff
@@ -33,7 +34,19 @@ Taking the first schema for reference, the updates look like that:
...
+section. To me it was hard to follow because ultimately what we want to say that all
+filtering features of the Algolia Engine (numeric, facets) are string or array string based
+and thus needs to be managed to allow easy manipulation like add, remove, toggle filter
+removing the need for the user to handle it himself.
+
+We can even give an example of clicking a facet value in a list with and without the helper.
+---
+
+Most of the parameters on the Algolia Rest API are simple: you just
+provide a string(?), a number (hitsPerPage).
```
Comment (bobylito) [inline scripts/docs/metalsmith/content/concepts.md]: string : queryType (query would have worked too but it has a shortcut).

File: scripts/docs/metalsmith/content/concepts.md [hunk 2]
```diff
@@ -46,8 +59,8 @@ input the string parameters, but this interface (that we call 'raw')
 is exclusive with the higher level one (that we call 'managed').
 
 This layer on top of the Algolia API even let us create features that don't
-exist in the normal API. We have created some features that rely on multiple
-queries based on the pattern we found over time.
+exist in the raw Algolia API. We have created some features that rely on multiple
```
Comment (bobylito) [inline scripts/docs/metalsmith/content/concepts.md]: I tried to use raw only for the parameters and configuration inside the helper, to avoid confusions. What about using Rest API here?

Comment (vvo) [inline scripts/docs/metalsmith/content/concepts.md]: Yes "raw API" => Rest API (or Algolia REST API)

File: scripts/docs/metalsmith/content/concepts.md (hunk 1 above)
Comment (vvo) [inline scripts/docs/metalsmith/content/concepts.md]: Ah yes in the end, query works well too. I don't even know what is queryType for

File: scripts/docs/metalsmith/content/concepts.md [hunk 3]
```diff
@@ -99,20 +112,33 @@ will be created.
...
+---
+Provide a simple text based menu like
+```
+- Products (200)
+  - fruits (100)
+    - citrus
+  - electronic (100)
+```
+---
+
```
Comment (bobylito) [inline scripts/docs/metalsmith/content/concepts.md]: Yep. I must say, I am a bit short when it comes to the hierachical facets.

File: scripts/docs/metalsmith/content/concepts.md (hunk 1 above)
Comment (vvo) [inline scripts/docs/metalsmith/content/concepts.md]: Even if there's a shortcut, still easy example in this context

File: scripts/docs/metalsmith/content/concepts.md [hunk 4]
```diff
@@ -99,20 +112,33 @@ will be created.
...
 When using the same high-tech e-shop example (see the previous paragraph about
 [disjunctive facetting](#disjunctive-facetting)), we can imagine that TV and video
 projectors are organized in a higher category called video hardware.
 
-The structure inside the record should contain all the variations per level. For a
+Since this is a feature built on top of the Algolia API, we expect the dataset
+to follow a hierachical notation (configurable).
```
Comment (bobylito) [inline scripts/docs/metalsmith/content/concepts.md]: Let's add a link to the documentation of the configuration.

File: scripts/docs/metalsmith/content/concepts.md [hunk 5]
```diff
@@ -99,20 +112,33 @@ will be created.
...
+
+By default, the structure inside the record should contain all the variations per level. For a
 product in the TV category, the record will have a `lvl1` attribute that will contain
 `video hardware` and a `lvl2` that will contain `video hardware > TV`. With the
 hierarchicalf feature, you can facet values per level and you will be provided the correct
 hierarchy of facet values.
 
-## Special capabilities
+## API style
```
Comment (bobylito) [inline scripts/docs/metalsmith/content/concepts.md]: I wanted to give a more fun feeling in this part. Those are the nice cool stuff, we should have a cool title name too ;)

File: scripts/docs/metalsmith/content/concepts.md [hunk 6]
```diff
@@ -99,20 +112,33 @@ will be created.
...
 
-What makes the special sauce of the Helper is a conjunction of patterns and
-smart tricks to help you make the best search UI.
+The helper is event based and is a chainable API much like jQuery.
```
Comment (bobylito) [inline scripts/docs/metalsmith/content/concepts.md]: Can we avoid the reference to jQuery? :)

Comment (vvo) [inline scripts/docs/metalsmith/content/concepts.md]: Eheheh when I put jQuery I thought "WOOPSIE!". Sure I think we can stop at "chainable API"

File: scripts/docs/metalsmith/content/concepts.md (hunk 5 above)
Comment (vvo) [inline scripts/docs/metalsmith/content/concepts.md]: Global API features?

I understand we are proud of those features but for a newcomer I won't see any excitement reading this :p

File: scripts/docs/metalsmith/content/concepts.md (hunk 6 above)
Comment (bobylito) [inline scripts/docs/metalsmith/content/concepts.md]: What about "The helper is more!" "MOARRRRRR" :D

**Comment by bobylito** [comment]
Thanks for comments :D
